openwisp_controller/config/static/config/js/widget.js

```diff
@@ -153,6 +153,29 @@
         return error;
     };
 
+    var validateMaxLength = function(){
+        var config_variables = {};
+        if (window.getContext()){
+            config_variables = JSON.parse(window.getContext().value);
+        }
+        var config_variables_keys = Object.keys(config_variables);
+        var all_fields = $(".jsoneditor .vTextField");
+        all_fields.map((index, field)=>{
+            if ($(field).attr('maxLength')){
+                var maxLength = $(field).attr('maxLength');
+                $(field).on('input', () => {
+                    $(config_variables_keys).map((i,key)=>{
```

Comment:
why are you looping over all the configuration variables?

Reply:
to check if the field value matches with the config_variables or not.

Reply:
Should I remove it @nemesisdesign ?

Reply:
I added this due to this https://github.com/openwisp/openwisp-controller/pull/410#discussion_r599992415